pkg/controller/jobframework/reconciler.go

```diff
@@ -109,9 +114,23 @@ func (r *JobReconciler) ReconcileGenericJob(ctx context.Context, req ctrl.Reques
 
 	// when manageJobsWithoutQueueName is disabled we only reconcile jobs that have either
 	// queue-name or the parent-workload annotation set.
-	if !r.manageJobsWithoutQueueName && QueueName(job) == "" && isStandaloneJob {
-		log.V(3).Info(fmt.Sprintf("Neither %s label, nor %s annotation is set, ignoring the job", QueueLabel, ParentWorkloadAnnotation))
-		return ctrl.Result{}, nil
+	// If the parent-workload annotation is set, it also checks whether the parent job has queue-name label.
+	if !r.manageJobsWithoutQueueName && QueueName(job) == "" {
+		if isStandaloneJob {
+			log.V(3).Info("Neither queue-name label, nor parent-workload annotation is set, ignoring the job",
+				controllerconsts.QueueLabel, QueueName(job), controllerconsts.ParentWorkloadAnnotation, ParentWorkloadName(job))
+			return ctrl.Result{}, nil
+		}
+		isParentJobManaged, err := r.isParentJobManaged(ctx, job.Object(), namespacedName.Namespace)
+		if err != nil {
+			log.Error(err, "couldn't check whether the parent job is managed by kueue")
+			return ctrl.Result{}, err
+		}
+		if !isParentJobManaged {
+			log.V(3).Info("parent-workload annotation is set, and the parent job doesn't have a queue-name label, ignoring the job",
+				controllerconsts.ParentWorkloadAnnotation, ParentWorkloadName(job))
```

Comment:
I would not use the same annotation, so that it's easier to write in a log query.

```suggestion
				"parentWorkload", ParentWorkloadName(job))
```

Reply:
I will fix L121 as well.

Reply:
In L294, we use `parent-workload`. So I will replace the annotation with `parent-workload` for consistency

```golang
			log.V(2).Info("job with no matching parent workload", "parent-workload", pwName)
```

Reply:
done.